src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/persist/SQLParsingUtilTest.java

```diff
@@ -0,0 +1,76 @@
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.persist;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.Lists;
+import org.jooq.Field;
+import org.jooq.Record;
+import org.jooq.Result;
+import org.jooq.exception.DataTypeException;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+public class SQLParsingUtilTest {
+    private static final String FIELD_NAME = "FIELD_NAME";
+    private static final String DATA_FIELD = "DATA_FIELD";
+
+    private SQLParsingUtil uut;
+
+    @Mock
+    private Result<Record> result;
+
+    @Mock
+    private Field<String> field;
+
+    @Mock
+    private Record record;
+
+    @Before
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+        uut = SQLParsingUtil.API;
+    }
+
+    @Test
+    public void testGetRecordByName() {
+        when(result.getValues(field)).thenReturn(Lists.newArrayList("no", FIELD_NAME, "nope"));
+        uut.getRecordByName(result, field, FIELD_NAME);
+        verify(result, times(1)).get(1);
+
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testGetRecordByNameWithNullResult() {
+        uut.getRecordByName(null, field, FIELD_NAME);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testGetRecordByNameWithNoMatch() {
+        when(result.getValues(field)).thenReturn(Lists.newArrayList("no", "matches", "here"));
+        uut.getRecordByName(result, field, FIELD_NAME);
+    }
+
+    @Test
+    public void testReadDataFromSqlResult() {
+        // method should return Double.NaN when result is null
+        assertEquals(Double.NaN, uut.readDataFromSqlResult(null, field, FIELD_NAME, DATA_FIELD), 0);
+        // method should return Double.NaN when it encounters an IllegalArgumentException
+        when(result.getValues(field)).thenReturn(Lists.newArrayList("no", "matches", "here"));
```

Comment:
I would suggest to create a dummy SQL context instead of mocking the Result object. You can use the jooq DSLcontext to create whatever Result<Record> you want and use that as input for this test.

Reply:
I disagree that this would be better than my current method of testing with mocks. I also think using the jooq DSL context would increase the overall line count of this test. Is there a strong reason to not use Mockito here?

Reply:
if we use mockito it would be hard to tell whether the exception "DataTypeException" can be emitted properly if we try to parse the result with wrong field name / data type. Anyway. I think it is a good to have and we can skip it for now.  :)